src/riak_kv_wm_queue.erl

```diff
@@ -77,25 +93,100 @@ service_available(RD, Ctx=#ctx{riak=RiakProps}) ->
+malformed_queuename(RD) ->
+    QueueNameRaw = wrq:path_info(queuename, RD),
+    list_to_atom(riak_kv_wm_utils:maybe_decode_uri(RD, QueueNameRaw)).
+
+-spec malformed_keyclocklist(iolist()) -> list(riak_kv_replrtq_src:repl_entry()).
+malformed_keyclocklist(ReqBody) ->
+    {struct, [{<<"keys-clocks">>, KCL}]} = mochijson2:decode(ReqBody),
+    lists:reverse(lists:foldl(fun decode_bucketkeyclock/2, [], KCL)).
+
+decode_bucketkeyclock({struct, BKC}, Acc) ->
+    B = 
+        case lists:keyfind(<<"bucket">>, 1, BKC) of
+            {<<"bucket">>, Bucket} when is_binary(Bucket) ->
+                case lists:keyfind(<<"bucket-type">>, 1, BKC) of
+                    {<<"bucket-type">>, BucketType} ->
+                        {BucketType, Bucket};
+                    false ->
+                        Bucket
+                end;
+            false ->
+                false
+        end,
+    case B of
+        false ->
+            Acc;
+        _ ->
+            case lists:keyfind(<<"key">>, 1, BKC) of
+                {<<"key">>, K} when is_binary(K) ->
+                    case lists:keyfind(<<"clock">>, 1, BKC) of
+                        {<<"clock">>, EncodedClock} ->
+                            DC =
+                                riak_object:decode_vclock(
+                                    base64:decode(EncodedClock)),
+                            [{B, K, DC}|Acc];
+                        false ->
+                            Acc
+                    end;
+                false ->
+                    Acc
+            end
+    end;
+decode_bucketkeyclock(_, Acc) ->
+    Acc.
 
 
 -spec content_types_provided(#wm_reqdata{}, context()) ->
     {[{ContentType::string(), Producer::atom()}], #wm_reqdata{}, context()}.
 %% @doc List the content types available for representing this resource.
-%%      "application/json" is the content-type for bucket lists.
-content_types_provided(RD, Ctx) ->
-    {[{"application/octet-stream", produce_queue_fetch}], RD, Ctx}.
+content_types_provided(RD, Ctx) when Ctx#ctx.method =:= 'GET' ->
+    {[{"application/octet-stream", produce_queue_fetch}], RD, Ctx};
+content_types_provided(RD, Ctx) when Ctx#ctx.method =:= 'POST' ->
+    {[{"text/plain", nop}], RD, Ctx}.
+
 
+-spec process_post(#wm_reqdata{}, context()) ->
+    {true, #wm_reqdata{}, context()}.
+%% @doc Pass-through for key-level requests to allow POST to function
+%%      as PUT for clients that do not support PUT.
+process_post(RD, Ctx) -> 
+    QueueName = Ctx#ctx.queuename,
+    KeyClockList = Ctx#ctx.keyclocklist,
+    ok = riak_kv_replrtq_src:replrtq_ttaaefs(QueueName, KeyClockList),
+    R = 
+        case riak_kv_replrtq_src:length_rtq(QueueName) of
+            {QueueName, {FL, FSL, RTL}} ->
+                io_lib:format("Queue ~w: ~w ~w ~w", [QueueName, FL, FSL, RTL]);
+            _ ->
+                io_lib:format("No queue ~w", [QueueName])
+        end,
+    {true,
+        wrq:set_resp_body(
```

Comment:
This is not very REST. One would expect to get the response as `application/json` instead of text. That means, instead of creating the io:format above, build a JSON structure.

Reply:
Acknowledged, but not fixed.